File: jvm/core/main/scala/com/webank/eggroll/core/deepspeed/job/meta/QueryJobStatusRequest.scala
package com.webank.eggroll.core.deepspeed.job.meta

import com.google.protobuf.ByteString
import com.webank.eggroll.core.meta.Deepspeed

import scala.language.implicitConversions

case class QueryJobStatusRequest(sessionId: String)

object QueryJobStatusRequest {
  implicit def serialize(src: QueryJobStatusRequest): Array[Byte] = {
    Deepspeed.QueryJobStatusRequest.newBuilder()
      .setSessionId(src.sessionId)
      .build()
      .toByteArray
  }

  implicit def deserialize(byteString: ByteString): QueryJobStatusRequest = {
    val proto = Deepspeed.QueryJobStatusRequest.parseFrom(byteString)
    QueryJobStatusRequest(sessionId = proto.getSessionId)
  }
}

File: jvm/new_core/src/main/java/com/eggroll/core/pojo/QueryJobStatusRequest.java
package com.eggroll.core.pojo;

import com.webank.eggroll.core.meta.Deepspeed;
import lombok.Data;
import org.slf4j.Logger;
import org.slf4j.LoggerFactory;

@Data
public class QueryJobStatusRequest implements RpcMessage{
    Logger log = LoggerFactory.getLogger(QueryJobStatusRequest.class);
    private String sessionId;

    @Override
    public byte[] serialize() {
        Deepspeed.QueryJobStatusRequest.Builder builder = Deepspeed.QueryJobStatusRequest.newBuilder();
        builder.setSessionId(this.sessionId);
        return builder.build().toByteArray();
    }

    @Override
    public void deserialize(byte[] data) {
        try {
            Deepspeed.QueryJobStatusRequest queryJobStatusRequest = Deepspeed.QueryJobStatusRequest.parseFrom(data);
            this.sessionId = queryJobStatusRequest.getSessionId();
        } catch (Exception e) {
            log.error("deserialize error : ", e);
        }
    }
}
